File: api/python/Impl/SysProvider.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
# SysProvider.py
# 2021, Lin Zhijun, https://github.com/toolgood/ToolGood.TextFilter.Api
# MIT Licensed 
 
__all__ = ['SysProvider']
import json
import http.client, urllib.parse
import requests


class SysProvider():

    def __init__(self,textFilterConfig):
        self.__textFilterConfig=textFilterConfig
        self.__UpdateSystemUrl = "/api/sys-update"
        self.__RefreshUrl = "/api/sys-refresh"
        self.__InfoUrl = "/api/sys-info"
        self.__UpdateLicenceUrl = "/api/sys-Update-Licence"
        self.__InitDataUrl = "/api/sys-init-Data"
        self.__GCCollectUrl = "/api/sys-GC-Collect"

    
    def UpdateSystem(self,textFilterNoticeUrl:str, textReplaceNoticeUrl:str,skipword:str):
        """     * 更新系统
      
        @textFilterNoticeUrl    默认 文本检测异步地址

        @textReplaceNoticeUrl   默认 文本替换异步地址

        @imageFilterNoticeUrl   默认 图片检测异步地址

        @imageClassifyNoticeUrl 默认 图片分类异步地址

        @imageTempPath          图片临时保存地址

        @skipword               自定义跳词

        """
        dictionary={
            'textFilterNoticeUrl':textFilterNoticeUrl,
            'textReplaceNoticeUrl':textReplaceNoticeUrl,
            'skipword':skipword
        }
        result=doPost(self.__textFilterConfig,self.__UpdateSystemUrl,dictionary)
        return json.loads(result)

    def Refresh(self):
        '刷新缓存'
        result=doGet(self.__textFilterConfig,self.__RefreshUrl)
        return json.loads(result)

    def Info(self):
        '产品信息'
        result=doGet(self.__textFilterConfig,self.__InfoUrl)
        return json.loads(result)

    def InitData(self):
        '重载数据'
        result=doGet(self.__textFilterConfig,self.__InitDataUrl)
        return json.loads(result)

    def GCCollect(self):
        'GC垃圾回收'
        result=doGet(self.__textFilterConfig,self.__GCCollectUrl)
        return json.loads(result)

 
 

def doGet(textFilterConfig,url):
    u=textFilterConfig.GetTextFilterHost()+url
    result = requests.get(u)
    return  result.text

def doPost(textFilterConfig,url,postData):
    u=textFilterConfig.GetTextFilterHost()+url
    result = requests.post(u, postData)
    return  result.text

File: api/golang/ToolGood/TextFilter/Api/Impl/SysProvider.go
package Impl

import "encoding/json"
import "net/http"
import "net/url"
import	"io/ioutil"
import	. "../Datas/Sys"
import	. "../Datas"

type SysProvider struct{
	textFilterHost string
}

func NewSysProvider(config string) *SysProvider{
	provider := &SysProvider{}
	provider.textFilterHost = config
	return  provider;
}

const updateSystemUrl string = "/api/sys-update";
const refreshUrl string = "/api/sys-refresh";
const infoUrl string = "/api/sys-info";
const updateLicenceUrl string= "/api/sys-Update-Licence";
const initDataUrl string = "/api/sys-init-Data";
const gcCollectUrl string = "/api/sys-GC-Collect";

func (this *SysProvider)UpdateSystem(textFilterNoticeUrl string,textReplaceNoticeUrl string,skipword string) *CommonResult  {
	u:=this.textFilterHost+updateSystemUrl;

	resp, err := http.PostForm(u, url.Values{ "textFilterNoticeUrl":{textFilterNoticeUrl}, "textReplaceNoticeUrl":{textReplaceNoticeUrl},"skipword":{skipword}	}) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *CommonResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}
func (this *SysProvider)Refresh() *CommonResult  {
	u:=this.textFilterHost+refreshUrl;
	return this.httpGet(u)
}
 
func (this *SysProvider)Info() *SysInfo  {
	u:=this.textFilterHost+infoUrl;
	resp, err := http.Get(u) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *SysInfo
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}
func (this *SysProvider)UpdateLicence(lic string) *CommonResult  {
	u:=this.textFilterHost+updateLicenceUrl;

	resp, err := http.PostForm(u, url.Values{ "Licence":{lic}}) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *CommonResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}

func (this *SysProvider)InitData() *CommonResult  {
	u:=this.textFilterHost+initDataUrl;
	return this.httpGet(u)
}
func (this *SysProvider)GCCollect() *CommonResult  {
	u:=this.textFilterHost+gcCollectUrl;
	return this.httpGet(u)
}






func (this *SysProvider)httpGet(u string) *CommonResult  {
	resp, err := http.Get(u) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *CommonResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}
